lib/asciidoctor-pdf/converter.rb

```diff
@@ -210,6 +210,12 @@ def convert_document doc
     font @theme.base_font_family, size: @theme.base_font_size, style: @theme.base_font_style.to_sym
     doc.set_attr 'pdf-anchor', (doc_anchor = derive_anchor_from_id doc.id, 'top')
     add_dest_for_block doc, doc_anchor
+    unless doc.doctype == 'book' || (doc.attr? 'title-page')
+      theme_font :heading, level: 1 do
+        align = (@theme.heading_h1_align || (doc.doctype == 'book' ? @theme.heading_align : :center) || @base_align).to_sym
+        layout_heading doc.doctitle, align: align, level: 1
+      end
+    end
```

Comment:
And if you move whole block for the article heading generation a bit up, like:

![image](https://user-images.githubusercontent.com/234774/51388869-c2fea700-1b3b-11e9-91d1-4620f99fb577.png)

so the toc creation would be after heading generation (the same as for book type) then we can have the following:


![image](https://user-images.githubusercontent.com/234774/51388751-6a2f0e80-1b3b-11e9-9e04-5a5a478ff0f3.png)

Now if there was an easy way to calculate vertical size of the heading (that might include subtitle and autours and revision date in the future) and shift toc down to that size...

Reply:
Pull request #986 already solves this (see also #233).
It has conflicts now, but I will do the merge after finishing #988.

Reply:
Let's address this in #988 since this PR has already been closed.